tests/reporter/kubernetes.go

```diff
@@ -173,6 +181,133 @@ func (r *KubernetesReporter) logVMIs(virtCli kubecli.KubevirtClient) {
 	fmt.Fprintln(f, string(j))
 }
 
+func (r *KubernetesReporter) logDMESG(virtCli kubecli.KubevirtClient, since time.Time) {
+
+	logsdir := filepath.Join(r.artifactsDir, "nodes")
+
+	if err := os.MkdirAll(logsdir, 0777); err != nil {
+		fmt.Fprintf(os.Stderr, "failed to create directory: %v\n", err)
+		return
+	}
+
+	vmis, err := virtCli.VirtualMachineInstance(v1.NamespaceAll).List(&v12.ListOptions{})
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to fetch vmis: %v\n", err)
+		return
+	}
+
+	nodes := map[string]struct{}{}
+	for _, vmi := range vmis.Items {
+		if vmi.Status.NodeName != "" {
+			nodes[vmi.Status.NodeName] = struct{}{}
+		}
+	}
+
+	rex := regexp.MustCompile(`\[([^]]+)]`)
+	for node, _ := range nodes {
+		func() {
+			f, err := os.OpenFile(filepath.Join(r.artifactsDir, "nodes", fmt.Sprintf("%d_dmesg_%s.log", r.failureCount, node)),
+				os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "failed to open the file: %v", err)
+				return
+			}
+			defer f.Close()
+			pod, err := kubecli.NewVirtHandlerClient(virtCli).Namespace(tests.KubeVirtInstallNamespace).ForNode(node).Pod()
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "failed to get virt-handler pod on node %s: %v", node, err)
+				return
+			}
+			// TODO may need to be improved, in case that the auditlog is really huge, since stdout is in memory
+			stdout, _, err := tests.ExecuteCommandOnPodV2(virtCli, pod, "virt-handler", []string{"/proc/1/root/bin/dmesg", "--kernel", "--ctime", "--userspace", "--decode"})
+			scanner := bufio.NewScanner(bytes.NewBufferString(stdout))
+			add := false
+			for scanner.Scan() {
+				if !add {
+					matches := rex.FindStringSubmatch(scanner.Text())
```

Comment:
You can put scanner.Text() in a variable so there you can reuse it later on

Reply:
Done.